dtm0/helper.c

```diff
@@ -0,0 +1,109 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_DTM0
+#include "lib/trace.h"
+#include "lib/assert.h"
+#include "lib/memory.h"
+#include "reqh/reqh_service.h"
+#include "reqh/reqh.h"               /* m0_reqh */
+#include "be/domain.h"
+#include "be/dtm0_log.h"
+#include "module/instance.h"         /* m0_get */
+
+
+static int dtm0_log_init(struct m0_be_domain *dom, const char *suffix,
+			 const struct m0_buf *data)
+{
+	struct m0_be_dtm0_log *log0 = *(struct m0_be_dtm0_log**)data->b_addr;
+	struct m0_reqh        *reqh = dom->bd_cfg.bc_engine.bec_reqh;
+	unsigned               key = m0_get()->i_dtm0_log_key;
+
+	M0_ENTRY("suffix: %s, data: %p, log0: %p", suffix, data->b_addr, log0);
+
+	if (m0_reqh_lockers_get(reqh, key) == NULL)
+		m0_reqh_lockers_set(reqh, key, log0);
+
+	return M0_RC(0);
```

Comment:
can be a void function or something more complex is going to be added here?

Reply:
.b0_init returns int therefore it an interface obligation.